### ISSUE TITLE
No permission for pg statefulset volume

### ISSUE THREAD

**Issue by lastonoga** [CrunchyData/crunchy-containers#287]
Created new statefulset with storageclass and volume. Everything creates but it does not work..

```
Mon Oct 16 14:51:35 UTC 2017
setting PGROOT to  /usr/pgsql-9.6
+ trap trap_sigterm SIGINT SIGTERM
+ date
+ source /opt/cpm/bin/setenv.sh
++ '[' -d /usr/pgsql-9.6 ']'
++ export PGROOT=/usr/pgsql-9.6
++ PGROOT=/usr/pgsql-9.6
++ echo 'setting PGROOT to ' /usr/pgsql-9.6
++ export PGDATA=/pgdata/pgset-0
++ PGDATA=/pgdata/pgset-0
++ export PGWAL=/pgwal/pgset-0-wal
++ PGWAL=/pgwal/pgset-0-wal
++ [[ -v PGDATA_PATH_OVERRIDE ]]
++ export PATH=/opt/cpm/bin:/usr/pgsql-9.6/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
++ PATH=/opt/cpm/bin:/usr/pgsql-9.6/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
++ export LD_LIBRARY_PATH=/usr/pgsql-9.6/lib
++ LD_LIBRARY_PATH=/usr/pgsql-9.6/lib
++ chown postgres /pgdata/pgset-0 /pgwal/pgset-0-wal
pgpguser does exist
chown: cannot access '/pgdata/pgset-0': No such file or directory
chown: cannot access '/pgwal/pgset-0-wal': No such file or directory
+ source check-for-secrets.sh
++ '[' -d /pguser ']'
++ echo 'pgpguser does exist'
+++ cat /pguser/username
++ export PG_USER=bloxy
++ PG_USER=bloxy
+++ cat /pguser/password
++ export 'PG_PASSWORD=KK&_frxRz++x'
++ PG_PASSWORD='KK&_frxRz++x'
++ '[' -d /pgroot ']'
++ echo 'pgroot does exist'
pgroot does exist
+++ cat /pgroot/password
++ export PG_ROOT_PASSWORD=qBSPp3P3x8VLiwScap61
++ PG_ROOT_PASSWORD=qBSPp3P3x8VLiwScap61
++ '[' -d /pgmaster ']'
++ echo 'pgmaster does exist'
+++ cat /pgmaster/username
pgmaster does exist
++ export PG_MASTER_USER=primary-user
++ PG_MASTER_USER=primary-user
+++ cat /pgmaster/password
++ export 'PG_MASTER_PASSWORD=tG9QH94F/H4Vbc&p'
++ PG_MASTER_PASSWORD='tG9QH94F/H4Vbc&p'
+ '[' '!' -v PG_MODE ']'
+ '[' set = slave ']'
+ '[' '!' -v PG_MASTER_USER ']'
+ '[' '!' -v PG_MASTER_PASSWORD ']'
+ '[' '!' -v PG_USER ']'
+ '[' '!' -v PG_PASSWORD ']'
+ '[' '!' -v PG_DATABASE ']'
+ '[' '!' -v PG_ROOT_PASSWORD ']'
+ export PG_MODE=set
+ PG_MODE=set
+ export PG_MASTER_HOST=
+ PG_MASTER_HOST=
+ export PG_MASTER_PORT=5432
+ PG_MASTER_PORT=5432
+ export PG_MASTER_USER=primary-user
+ PG_MASTER_USER=primary-user
+ export 'PG_MASTER_PASSWORD=tG9QH94F/H4Vbc&p'
+ PG_MASTER_PASSWORD='tG9QH94F/H4Vbc&p'
+ export PG_USER=bloxy
+ PG_USER=bloxy
+ export 'PG_PASSWORD=KK&_frxRz++x'
+ PG_PASSWORD='KK&_frxRz++x'
+ export PG_DATABASE=bloxy
+ PG_DATABASE=bloxy
+ export PG_ROOT_PASSWORD=qBSPp3P3x8VLiwScap61
+ PG_ROOT_PASSWORD=qBSPp3P3x8VLiwScap61
+ mkdir -p /pgdata/pgset-0
mkdir: cannot create directory '/pgdata/pgset-0': Permission denied
+ chmod 0700 /pgdata/pgset-0
chmod: cannot access '/pgdata/pgset-0': No such file or directory
+ [[ -v ARCHIVE_MODE ]]
+ rm /pgdata/pgset-0/postmaster.pid
rm: cannot remove '/pgdata/pgset-0/postmaster.pid': No such file or directory
+ echo 'user id is...'
+ id
+ case "$PG_MODE" in
+ role_discovery
+ PATH=/opt/cpm/bin:/usr/pgsql-9.6/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/cpm/bin
+ ordinal=0
+ echo 0 is ordinal
+ '[' 0 -eq 0 ']'
+ kubectl label --overwrite=true pod pgset-0 name=
user id is...
uid=26(postgres) gid=26(postgres) groups=26(postgres)
0 is ordinal
pod "pgset-0" labeled
+ oc label --overwrite=true pod pgset-0 name=
pod "pgset-0" not labeled
setting PG_MODE to master
+ echo 'setting PG_MODE to master'
+ export PG_MODE=master
+ PG_MODE=master
+ ose_hack
++ id -u
+ export USER_ID=26
+ USER_ID=26
++ id -g
+ export GROUP_ID=26
+ GROUP_ID=26
+ envsubst
+ export LD_PRELOAD=/usr/lib64/libnss_wrapper.so
+ LD_PRELOAD=/usr/lib64/libnss_wrapper.so
+ export NSS_WRAPPER_PASSWD=/tmp/passwd
+ NSS_WRAPPER_PASSWD=/tmp/passwd
+ export NSS_WRAPPER_GROUP=/etc/group
+ NSS_WRAPPER_GROUP=/etc/group
+ fill_conf_file
+ [[ -v TEMP_BUFFERS ]]
+ TEMP_BUFFERS=8MB
+ [[ -v MAX_CONNECTIONS ]]
+ MAX_CONNECTIONS=100
+ [[ -v SHARED_BUFFERS ]]
+ SHARED_BUFFERS=128MB
+ [[ -v WORK_MEM ]]
+ WORK_MEM=4MB
+ [[ -v MAX_WAL_SENDERS ]]
+ MAX_WAL_SENDERS=6
+ cp /opt/cpm/conf/postgresql.conf.template /tmp/postgresql.conf
+ [[ -v ARCHIVE_MODE ]]
+ ARCHIVE_MODE=off
+ [[ -v ARCHIVE_TIMEOUT ]]
+ ARCHIVE_TIMEOUT=0
+ '[' -f /pgconf/pgbackrest.conf ']'
+ sed -i s/TEMP_BUFFERS/8MB/g /tmp/postgresql.conf
+ sed -i s/MAX_CONNECTIONS/100/g /tmp/postgresql.conf
working on master...
initialize_master
pgdata is empty and id is...
uid=26(postgres) gid=26(postgres) groups=26(postgres)
+ sed -i s/SHARED_BUFFERS/128MB/g /tmp/postgresql.conf
+ sed -i s/MAX_WAL_SENDERS/6/g /tmp/postgresql.conf
+ sed -i s/WORK_MEM/4MB/g /tmp/postgresql.conf
+ sed -i s/ARCHIVE_MODE/off/g /tmp/postgresql.conf
+ sed -i s/ARCHIVE_TIMEOUT/0/g /tmp/postgresql.conf
+ case "$PG_MODE" in
+ echo 'working on master...'
+ initialize_master
+ echo initialize_master
+ '[' '!' -f /pgdata/pgset-0/postgresql.conf ']'
+ echo 'pgdata is empty and id is...'
+ id
checking_for_restore
total 0
no backup file found...
doing initdb....
initdb -D /pgdata/pgset-0 > /tmp/initdb.log &> /tmp/initdb.err
+ mkdir -p /pgdata/pgset-0
mkdir: cannot create directory '/pgdata/pgset-0': Permission denied
+ check_for_restore
+ echo checking_for_restore
+ ls -l /backup
+ '[' '!' -f /backup//postgresql.conf ']'
+ echo 'no backup file found...'
+ initdb_logic
+ echo 'doing initdb....'
+ cmd='initdb -D /pgdata/pgset-0 '
+ [[ -v PG_LOCALE ]]
+ [[ -v XLOGDIR ]]
+ [[ -v CHECKSUMS ]]
+ cmd+=' > /tmp/initdb.log &> /tmp/initdb.err'
+ echo initdb -D /pgdata/pgset-0 '>' /tmp/initdb.log '&>' /tmp/initdb.err
+ eval initdb -D /pgdata/pgset-0 '>' /tmp/initdb.log '&>' /tmp/initdb.err
++ initdb -D /pgdata/pgset-0
+ echo 'overlay pg config with your settings....'
+ cp /tmp/postgresql.conf /pgdata/pgset-0
cp: cannot create regular file '/pgdata/pgset-0': Permission denied
+ cp /opt/cpm/conf/pg_hba.conf /tmp
+ sed -i s/PG_MASTER_USER/primary-user/g /tmp/pg_hba.conf
overlay pg config with your settings....
+ cp /tmp/pg_hba.conf /pgdata/pgset-0
cp: cannot create regular file '/pgdata/pgset-0': Permission denied
+ check_for_pitr
+ echo 'checking for PITR WAL files to recover with..'
++ ls -A /recover
checking for PITR WAL files to recover with..
temporarily starting db to run setup.sql
+ '[' '' ']'
+ echo 'starting db'
+ echo 'temporarily starting db to run setup.sql'
+ pg_ctl -D /pgdata/pgset-0 start
pg_ctl: directory "/pgdata/pgset-0" does not exist
+ echo 'waiting for pg to start up...'
+ true
+ pg_isready --port=5432 --host=pgset-0 --username=primary-user --timeout=2
+ '[' 2 -eq 0 ']'
+ sleep 2
waiting for pg to start up...
pgset-0:5432 - no response
pgset-0:5432 - no response
+ true
+ pg_isready --port=5432 --host=pgset-0 --username=primary-user --timeout=2
+ '[' 2 -eq 0 ']'
+ sleep 2
+ true
+ pg_isready --port=5432 --host=pgset-0 --username=primary-user --timeout=2
+ '[' 2 -eq 0 ']'
+ sleep 2
pgset-0:5432 - no response

```

```
{
    "apiVersion": "apps/v1beta1",
    "kind": "StatefulSet",
    "metadata": {
        "name": "pgset"
    },
    "spec": {
        "serviceName": "pgset",
        "replicas": 2,
        "template": {
            "metadata": {
                "labels": {
                    "app": "pgset"
                }
            },
            "spec": {
            	"terminationGracePeriodSeconds": 10,
                "containers": [{
                    "name": "pgset",
                    "image": "crunchydata/crunchy-postgres:$CCP_IMAGE_TAG",
                    "ports": [{
                        "containerPort": 5432,
                        "name": "postgres"
                    }],
                    "env": [{
                        "name": "PG_MASTER_PORT",
                        "value": "5432"
                    }, {
                        "name": "PGHOST",
                        "value": "/tmp"
                    }, {
                        "name": "PG_MODE",
                        "value": "set"
                    }, {
                        "name": "PG_DATABASE",
                        "value": "bloxy"
                    }, {
                    	"name": "PG_MASTER_USER",
                    	"value": "/pgprimary/username"
                    }, {
                    	"name": "PG_MASTER_PASSWORD",
                    	"value": "/pgprimary/password"
                    }],
                    "volumeMounts": [{
                        "mountPath": "/pgdata",
                        "name": "pgdata",
                        "readOnly": false
                    }, {
                        "mountPath": "/pguser",
                        "name": "pguser-volume"
                    }, {
                        "mountPath": "/pgmaster",
                        "name": "pgprimary-volume"
                    }, {
                        "mountPath": "/pgroot",
                        "name": "pgroot-volume"
                    }]
                }],
                "volumes": [{
					"name": "pguser-volume",
					"secret": {
						"secretName": "pguser-secret"
					}
				}, {
					"name": "pgprimary-volume",
					"secret": {
						"secretName": "pgprimary-secret"
					}
				}, {
					"name": "pgroot-volume",
					"secret": {
						"secretName": "pgroot-secret"
					}
				}]

            }
        },
        "volumeClaimTemplates": [{
            "metadata": {
                "name": "pgdata"
            },
            "spec": {
            	"storageClassName": "fast",
                "accessModes": [
                    "ReadWriteOnce"
                ],
                "resources": {
                    "requests": {
                        "storage": "10Gi"
                    }
                }
            }
        }]
    }
}

```

env
```
export CCP_CLI=kubectl
export CCP_IMAGE_TAG=centos7-9.6-1.5.1
export NAMESPACE=default
#export NAMESPACE=demo-project
export PV_PATH=/pgdata
```

Also **there are a lot of** failing description in documentation.

Pod did not want to be create cause of "no PG_MASTER_USER data"
unless i added it to the yaml.

And there is no information about PG_MASTER_USER variable. 
Also there are no information about which type of volume should we use. Can we use ReadWriteOnce volume for increasing high availability.

**Comment by jmccormick2001** [comment]
probably what has happened is that you are working off of master which is in flux at the moment...we are pushing release 1.6.0 out tonight which builds off of master, there are changes with this new release that will require you to modify your example to the latest version...you might try it again tomorrow with the new images on dockerhub.

**Comment by xenophenes** [comment]
I experienced these same issues; what solved it for me was installing the postgresql-server package and disabling SELinux. Please give that a try and let me know if the issue persists.

**Comment by xenophenes** [comment]
Hello! Are these issues still persisting for you? Have you had a chance to experiment with the new v1.6.0 release?

**Comment by xenophenes** [comment]
@jmccormick2001 I'd say this issue is good to close.

**Comment by lastonoga** [comment]
@xenophenes Sorry for long answer. I haven't tested that yet, cause i switched to another solution -
 stolon

**Comment by ksatchit** [comment]
@jmccormick2001 @xenophenes , at the outset, thanks for coming up with crunchy containers ! 

I am facing the same issue where all pods and services are created but the db init fails with permission denied errors on pgdata. And this is with the 1.6.0 release (image: centos7-10.0-1.6.0).  The set.json needed "PG_PRIMARY_PORT" (this was alluded to in the 1.6.0 release notes) without which the pods do not come up. 

The nodes are ubuntu 16.04 VMs, and SELinux is not available by default. However, I have disabled apparmor. Also, installing the postgres-database server package before deployment didn't help. 

Could you please look into this/suggest a solution ? 

Provided below are environment info, the startup logs & initdb_err.log from pgset-0 pod and the set.json specification file.

ENVIRONMENT: 
------------------
Kubernetes Hosts : Ubuntu 16.04 x64 ESX VMs
Kubernetes Version: 1.7.7 (setup via kubeadm) 
Crunchy Container: crunchydata/crunchy-postgres:centos7-10.0-1.6.0
PersistentVolume : OpenEBS PVs (via openebs storage class & dynamic provisioner) 

PGSET-0 STARTUP LOG
--------------------------
```
Mon Nov 13 01:39:25 UTC 2017
+ trap trap_sigterm SIGINT SIGTERM
setting PGROOT to  /usr/pgsql-10
+ date
+ source /opt/cpm/bin/setenv.sh
++ '[' -d /usr/pgsql-10 ']'
++ export PGROOT=/usr/pgsql-10
++ PGROOT=/usr/pgsql-10
++ echo 'setting PGROOT to ' /usr/pgsql-10
++ export PGDATA=/pgdata/pgset-0
++ PGDATA=/pgdata/pgset-0
++ export PGWAL=/pgwal/pgset-0-wal
++ PGWAL=/pgwal/pgset-0-wal
++ [[ -v PGDATA_PATH_OVERRIDE ]]
++ export PATH=/opt/cpm/bin:/usr/pgsql-10/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
++ PATH=/opt/cpm/bin:/usr/pgsql-10/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
++ export LD_LIBRARY_PATH=/usr/pgsql-10/lib
++ LD_LIBRARY_PATH=/usr/pgsql-10/lib
++ chown postgres /pgdata/pgset-0 /pgwal/pgset-0-wal
chown: cannot access '/pgdata/pgset-0': No such file or directory
chown: cannot access '/pgwal/pgset-0-wal': No such file or directory
+ source check-for-secrets.sh
++ '[' -d /pguser ']'
++ '[' -d /pgroot ']'
++ '[' -d /pgprimary ']'
+ '[' '!' -v PG_MODE ']'
+ '[' primary = master ']'
+ '[' primary = slave ']'
+ '[' -v PG_MASTER_HOST ']'
+ '[' -v PG_MASTER_USER ']'
+ '[' -v PG_MASTER_PASSWORD ']'
+ '[' -v PG_MASTER_PORT ']'
+ '[' primary = replica ']'
+ [[ ! -v PG_PRIMARY_USER ]]
+ [[ ! -v PG_PRIMARY_PASSWORD ]]
+ '[' '!' -v PG_USER ']'
+ '[' '!' -v PG_PASSWORD ']'
+ '[' '!' -v PG_DATABASE ']'
+ '[' '!' -v PG_ROOT_PASSWORD ']'
+ [[ ! -v PG_PRIMARY_PORT ]]
+ export PG_MODE=primary
+ PG_MODE=primary
+ export PG_PRIMARY_HOST=
+ PG_PRIMARY_HOST=
+ export PG_PRIMARY_PORT=5432
+ PG_PRIMARY_PORT=5432
+ export PG_PRIMARY_USER=primaryuser
+ PG_PRIMARY_USER=primaryuser
+ export PG_PRIMARY_PASSWORD=password
+ PG_PRIMARY_PASSWORD=password
+ export PG_USER=testuser
+ PG_USER=testuser
+ export PG_PASSWORD=password
+ PG_PASSWORD=password
+ export PG_DATABASE=userdb
+ PG_DATABASE=userdb
+ export PG_ROOT_PASSWORD=password
+ PG_ROOT_PASSWORD=password
+ mkdir -p /pgdata/pgset-0
mkdir: cannot create directory '/pgdata/pgset-0': Permission denied
+ chmod 0700 /pgdata/pgset-0
chmod: cannot access '/pgdata/pgset-0': No such file or directory
+ [[ -v ARCHIVE_MODE ]]
+ rm /pgdata/pgset-0/postmaster.pid
User id is...
uid=26(postgres) gid=26(postgres) groups=26(postgres)
rm: cannot remove '/pgdata/pgset-0/postmaster.pid': No such file or directory
+ echo 'User id is...'
+ id
+ case "$PG_MODE" in
+ ose_hack
++ id -u
+ export USER_ID=26
+ USER_ID=26
++ id -g
+ export GROUP_ID=26
+ GROUP_ID=26
+ envsubst
+ export LD_PRELOAD=/usr/lib64/libnss_wrapper.so
+ LD_PRELOAD=/usr/lib64/libnss_wrapper.so
+ export NSS_WRAPPER_PASSWD=/tmp/passwd
+ NSS_WRAPPER_PASSWD=/tmp/passwd
+ export NSS_WRAPPER_GROUP=/etc/group
+ NSS_WRAPPER_GROUP=/etc/group
+ fill_conf_file
+ [[ -v TEMP_BUFFERS ]]
+ TEMP_BUFFERS=8MB
+ [[ -v MAX_CONNECTIONS ]]
+ MAX_CONNECTIONS=100
+ [[ -v SHARED_BUFFERS ]]
+ SHARED_BUFFERS=128MB
+ [[ -v WORK_MEM ]]
+ WORK_MEM=4MB
+ [[ -v MAX_WAL_SENDERS ]]
+ MAX_WAL_SENDERS=6
+ cp /opt/cpm/conf/postgresql.conf.template /tmp/postgresql.conf
+ [[ -v ARCHIVE_MODE ]]
+ ARCHIVE_MODE=off
+ [[ -v ARCHIVE_TIMEOUT ]]
+ ARCHIVE_TIMEOUT=0
+ '[' -f /pgconf/pgbackrest.conf ']'
+ sed -i s/TEMP_BUFFERS/8MB/g /tmp/postgresql.conf
+ sed -i s/MAX_CONNECTIONS/100/g /tmp/postgresql.conf
+ sed -i s/SHARED_BUFFERS/128MB/g /tmp/postgresql.conf
+ sed -i s/MAX_WAL_SENDERS/6/g /tmp/postgresql.conf
+ sed -i s/WORK_MEM/4MB/g /tmp/postgresql.conf
+ sed -i s/ARCHIVE_MODE/off/g /tmp/postgresql.conf
+ sed -i s/ARCHIVE_TIMEOUT/0/g /tmp/postgresql.conf
+ case "$PG_MODE" in
+ echo 'Working on primary...'
Working on primary...
+ initialize_primary
+ echo initialize_primary
initialize_primary
+ '[' '!' -f /pgdata/pgset-0/postgresql.conf ']'
+ echo 'pgdata is empty and id is...'
pgdata is empty and id is...
+ id
uid=26(postgres) gid=26(postgres) groups=26(postgres)
+ mkdir -p /pgdata/pgset-0
mkdir: cannot create directory '/pgdata/pgset-0': Permission denied
+ check_for_restore
+ echo checking_for_restore
checking_for_restore
+ ls -l /backup
total 0
+ '[' '!' -f /backup//postgresql.conf ']'
+ echo 'No backup file found...'
No backup file found...
+ initdb_logic
+ echo 'Doing initdb...'
Doing initdb...
+ cmd='initdb -D /pgdata/pgset-0 '
+ [[ -v PG_LOCALE ]]
+ [[ -v XLOGDIR ]]
+ [[ -v CHECKSUMS ]]
+ cmd+=' > /tmp/initdb.log &> /tmp/initdb.err'
+ echo initdb -D /pgdata/pgset-0 '>' /tmp/initdb.log '&>' /tmp/initdb.err
initdb -D /pgdata/pgset-0 > /tmp/initdb.log &> /tmp/initdb.err
+ eval initdb -D /pgdata/pgset-0 '>' /tmp/initdb.log '&>' /tmp/initdb.err
++ initdb -D /pgdata/pgset-0
+ echo 'Overlaying postgreSQL'\''s configuration with your settings...'
Overlaying postgreSQL's configuration with your settings...
+ cp /tmp/postgresql.conf /pgdata/pgset-0
cp: cannot create regular file '/pgdata/pgset-0': Permission denied
+ cp /opt/cpm/conf/pg_hba.conf /tmp
+ sed -i s/PG_PRIMARY_USER/primaryuser/g /tmp/pg_hba.conf
+ cp /tmp/pg_hba.conf /pgdata/pgset-0
cp: cannot create regular file '/pgdata/pgset-0': Permission denied
+ check_for_pitr
+ echo 'Checking for PITR WAL files to recover with...'
Checking for PITR WAL files to recover with...
++ ls -A /recover
+ '[' '' ']'
+ echo 'Starting database...'
+ echo 'Temporarily starting database to run setup.sql...'
Temporarily starting database to run setup.sql...
+ pg_ctl -D /pgdata/pgset-0 start
pg_ctl: directory "/pgdata/pgset-0" does not exist
+ echo 'Waiting for postgreSQL to start...'
Waiting for postgreSQL to start...
+ true
+ pg_isready --port=5432 --host=pgset-0 --username=primaryuser --timeout=2
pgset-0:5432 - no response
+ '[' 2 -eq 0 ']'
+ sleep 2
+ true
+ pg_isready --port=5432 --host=pgset-0 --username=primaryuser --timeout=2
pgset-0:5432 - no response
+ '[' 2 -eq 0 ']'
+ sleep 2
```

INITDB_ERR.LOG: 
-------------------
```
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locale "C".
The default database encoding has accordingly been set to "SQL_ASCII".
The default text search configuration will be set to "english".

Data page checksums are disabled.

creating directory /pgdata/pgset-0 ... initdb: could not create directory "/pgdata/pgset-0": Permission denied
```
SET.JSON:
-----------
```
{
  "apiVersion": "apps/v1beta1",
  "kind": "StatefulSet",
  "metadata": {
    "name": "pgset"
  },
  "spec": {
    "serviceName": "pgset",
    "replicas": 2,
    "template": {
      "metadata": {
        "labels": {
          "app": "pgset"
        }
      },
      "spec": {
        "containers": [
          {
            "name": "pgset",
            "image": "crunchydata/crunchy-postgres:centos7-10.0-1.6.0",
            "ports": [
              {
                "containerPort": 5432,
                "name": "postgres"
              }
            ],
            "env": [{
                "name": "PG_PRIMARY_USER",
                "value": "primaryuser"
            }, {
                "name": "PGHOST",
                "value": "/tmp"
            }, {
                "name": "PG_MODE",
                "value": "primary"
            }, {
                "name": "PG_PRIMARY_PASSWORD",
                "value": "password"
            }, {
                "name": "PG_USER",
                "value": "testuser"
            }, {
                "name": "PG_PASSWORD",
                "value": "password"
            }, {
                "name": "PG_DATABASE",
                "value": "userdb"
            }, {
                "name": "PG_ROOT_PASSWORD",
                "value": "password"
            }, {
                "name": "PG_PRIMARY_PORT",
                "value": "5432"
            }],
            "volumeMounts": [
              {
                "name": "pgdata",
                "mountPath": "/pgdata",
		"readOnly": false
              }
            ]
          }
        ]
      }
    },
    "volumeClaimTemplates": [
      {
        "metadata": {
          "name": "pgdata"
        },
        "spec": {
          "accessModes": [
            "ReadWriteOnce"
          ],
	  "storageClassName": "openebs-standard",
          "resources": {
            "requests": {
              "storage": "400M"
            }
          }
        }
      }
    ]
  }
}
```

**Comment by jmccormick2001** [comment]
try adding this to your set.json

 "spec": {
            "securityContext": {
                "fsGroup": 26
            },
                "containers": [{

**Comment by ksatchit** [comment]
Thanks @jmccormick2001 !

The database init was successful upon adding the securityContext. Found that I needed to make a few more minor changes to the spec to get the replication cluster working successfully. 

In all, I have made the following changes to the set.json specified here :  https://github.com/CrunchyData/crunchy-containers/blob/master/examples/kube/statefulset-dyn/set.json

- Add securityContext with fsgroup: 26
- Change PG_MODE value to "set" instead of "primary"
- Add PG_PRIMARY_PORT with value set to 5432
- Add PG_PRIMARY_HOST with value set to "pgset-primary"

Raised a PR #324 to add these changes to the spec.
